### PR TITLE
--env-host: use default from containers.conf

### PR DESCRIPTION
**Pull request by vrothberg**
As found while working on #20000, the `--env-host` flag should use the default from containers.conf.  Add a new "supported fields" test to the system tests to make sure we have a goto test for catching such regressions.  I suspect more flags to not use the defaults from containers.conf.

#### Does this PR introduce a user-facing change?

```release-note
The default value of --env-host can now be configured via containers.conf.
```

### CODE REVIEW

File: test/system/800-config.bats [hunk 1]
```diff
@@ -91,6 +91,7 @@ EOF
     cat > $conf_tmp <<EOF
 [containers]
 annotations=['module=$random_data']
+env_host=true
```
Comment (Luap99) [inline test/system/800-config.bats:94]: Why this change? Does not look like you test this anywhere.

File: cmd/podman/common/create.go [hunk 2]
```diff
@@ -140,7 +140,9 @@ func DefineCreateFlags(cmd *cobra.Command, cf *entities.ContainerCreateOptions,
 		if !registry.IsRemote() {
 			createFlags.BoolVar(
 				&cf.EnvHost,
-				"env-host", false, "Use all current host environment variables in container",
+				"env-host",
+				podmanConfig.ContainersConfDefaultsRO.Containers.EnvHost,
```
Comment (Luap99) [inline cmd/podman/common/create.go:144]: This is not the the best (I would even say correct) place to set defaults.

Mainly this will bypass the server defaults, so it will not take effect for users that use the API.

File: test/system/800-config.bats (hunk 1 above)
Comment (vrothberg) [reply test/system/800-config.bats:94]: Artifact...I first wanted to move the check into this test.  Thanks for catching.

File: cmd/podman/common/create.go (hunk 2 above)
Comment (rhatdan) [reply cmd/podman/common/create.go:144]: The change is not for remote. So it does not bypass server, since it does not work with remote.

The difficulty difference between --remote and --remote=false is when you execute `podman CMD --help` cobra is listing the defaults.  In the local case we can figure out the current defaults from containers.conf, in the --remote case we can not figure these defaults so we are forced to just list what we got.

Comment (vrothberg) [reply cmd/podman/common/create.go:144]: @rhatdan what Paul pointed at is that `podman system service` is not loading these defaults, so clients (e.g., podman-py) will not notice any changes when they configure containers.conf.  That's a fart in the REST API and how we use containers.conf

But getting this done (IMO) is outside of the context of this PR and probably quite some work.

Comment (rhatdan) [reply cmd/podman/common/create.go:144]: Agreed.

**Comment by rhatdan** [comment]
LGTM
